### ISSUE TITLE
SIGABRT with `bun --watch` and `chokidar.watch`

### ISSUE THREAD

**Issue by eliot-akira** [oven-sh/bun#9031]
### What steps can reproduce the bug?

I encountered a crash, "terminated by signal SIGABRT", apparently arising from an interaction between `bun --watch`, `chokidar.watch`, and `fs.exists`.

It happens every time with the following snippet.

```ts
import fs from 'node:fs/promises'
import chokidar from 'chokidar'

const fileName = 'index.html'

chokidar.watch(fileName) // Doesn't crash if commented out

const server = Bun.serve({
  async fetch(request, server) {

    const file = Bun.file(fileName)

    if (
      !await fs.exists(fileName) // Crashes if this is here
      // !await file.exists()    // But this is OK
    ) {
      return new Response('not found', {
        status: 404,
      })
    }

    return new Response(file, {
      headers: { 'Content-Type': 'text/html' },
    })
  },
})

console.log(`See http://localhost:${server.port}`)
```

Some notable points:

- It only happens with `bun run --watch`, not `bun run`
- If `chokidar.watch` is commented out, it doesn't crash
- If `fs.exists` is commented out, it doesn't crash
- It still happens if `chokidar` and `fs` are accessing different files

---

I prepared a [repository](https://github.com/eliot-akira/bun-watch-chokidar-issue) that demonstrates the issue.

```sh
git clone https://github.com/eliot-akira/bun-watch-chokidar-issue
cd bun-watch-chokidar-issue
bun install
bun run start

# Visit http://localhost:3000

error: script "start" was terminated by signal SIGABRT (Abort)
Aborted (core dumped)
```

### What version of Bun is running?

1.0.28+705638470

### What platform is your computer?

Linux 6.4.6-76060406-generic x86_64 x86_64

**Comment by Jarred-Sumner** [comment]
This was fixed in Bun v1.0.29

**Comment by sangcodelynx** [comment]
I meet this issue again in Bun v1.0.38
